agent.go

```diff
@@ -896,44 +959,20 @@ func (p *pod) runContainerProcess(cid, pid string, terminal bool, started chan e
 
 	started <- nil
 
-	processState, err := proc.process.Wait()
-	// Ignore error if process fails because of an unsuccessful exit code
-	if _, ok := err.(*exec.ExitError); err != nil && !ok {
-		fieldLogger.WithError(err).Error("Process wait failed")
+	// Using helper function wait() to deal with the subreaper.
+	libContProcess := (*reaperLibcontainerProcess)(&(proc.process))
+	exitCode, err := p.subreaper.wait(processID, libContProcess)
+	if err != nil {
+		return err
 	}
-	// Close pipes to terminate routeOutput() go routines.
-	ctr.closeProcessPipes(pid)
```

Comment:
this is not required anymore?

Reply:
This is still called from the defer here: https://github.com/clearcontainers/agent/pull/178/files/e8d8a5fc6359f172177a782b85f153011d008015..c36a44adc6e06215b1c4cb736aa1e5d57699f82f#diff-2fe34e1b6f7f9aece0af74b1635502eeR889